packages/block-editor/src/components/text-orientation-control/index.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { textMixed, textUpright } from '@wordpress/icons';
+
+const TEXT_ORIENTATIONS = [
+	{
+		name: __( 'Upright' ),
+		value: 'upright',
+		icon: textUpright,
+	},
+	{
+		name: __( 'Mixed' ),
+		value: 'mixed',
+		icon: textMixed,
+	},
+];
+
+/**
+ * Control to facilitate text orientation selections.
+ *
+ * @param {Object}   props           Component props.
+ * @param {string}   props.className Class name to add to the control.
+ * @param {string}   props.value     Currently selected text orientation.
+ * @param {Function} props.onChange  Handles change in the text orientation selection.
+ *
+ * @return {WPElement} Text Orientation control.
+ */
+export default function TextOrientationControl( {
+	className,
+	value,
+	onChange,
+} ) {
+	return (
+		<fieldset
+			className={ classnames(
+				'block-editor-text-orientation-control',
+				className
+			) }
+		>
+			<BaseControl.VisualLabel as="legend">
+				{ __( 'Character rotation' ) }
+			</BaseControl.VisualLabel>
```

Comment:
Check if there something else we can use other than `visualLabel` since it is visually hidden with the CSS.